File: src/main/scala/org/mahedi/scala/examples/ArraySuite.scala
package org.mahedi.scala.examples

import scala.util.Sorting

/**
 * 
 * @author mahedi
 *
 */
object ArraySuite {

  /**
   * 
   */
  def getMaxValue(values: Array[Int]): Int = {
    if (values == null) throw new IllegalArgumentException()
    Sorting.quickSort(values)
    return values(values.length - 1)
  }

  def main(args: Array[String]): Unit = {
    //getMaxValue
    val values = Array(4, 3, 7, 5, 9, 0)
    println(ArraySuite.getMaxValue(values));

  }
}

File: src/main/java/org/mahedi/java/examples/ArraySuite.java
package org.mahedi.java.examples;

import java.util.Arrays;

/**
 * 
 * @author mahedi
 *
 */
public class ArraySuite {
	/**
	 * 
	 * @param values
	 * @return max integer value
	 */
	public static int getMaxValue(int[] values) {
		if (values == null)
			throw new IllegalArgumentException();
		Arrays.sort(values);
		return values[values.length - 1];
	}

	public static void main(String[] args) {
		int[] values = {4,3,7,5,9,0};
		System.out.println(ArraySuite.getMaxValue(values));
	}
}
